### PR TITLE
Issue-51: Movie form autopopulation when adding to a List

### LINKED ISSUE #51: Movie form autopopulation when adding to a List
Instead of relying on the users to enter their own movies to their lists it'd be a better approach to pull movies from an existing api of movies and populate it that way, thus ensuring only clean data is ever entered into the db.

[TMDB](https://developer.themoviedb.org/docs) seems to solve this very problem. I think a form could be created with a dropdown list and upon each debounced keystroke a new film would appear to populate the dropdown list. Users must click on the movie they want to add to their list and have the form autopopulate the remainder of the entry. The rest of the form (description, genre, and release year) should be filled out by the api- I imagine those form fields can contain the data but be locked from editing? Not sure here.

@Thinbishop had mentioned interest in figuring this out via the NextJS server side. Should be a lengthy process but a very worthy one.

### PR DESCRIPTION
**Pull request by Thinbishop**
## Issue

Closes #51

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description

Currently, movies need to be manually added and their details populated in a movie list. Integrating [TMDB](www.themoviedb.org) as a data source will allow us to populate movie details automatically.

When editing a movie list, the user can click to add a movie, then search for their movie and once found, they can select it. The movie will be automatically added to the movie list. 

This also changes the editability of the movie list `<Datagrid/>` component, and required removal of the `<GenreDropdown/>` in favour of a `<GenreDisplay/>`, in which the chips are displayed, without a dropdown selection menu.

Additionally, tool tip styling has been changed in `theme.ts` . The background color has been changed to black at 70% transparency.

lodash has been added as a dependency to implement search text debounce.

## Testing the PR

- run backend as usual
- `cd web` -> `npm i` to install lodash
- `npm run dev`
- Create a list and go to its page
- Click the ➕ button on the list to open the add movie dialog
- Search for and add various movies. 
  - These should be added to the list, but not saved.
  - To save, click the list save icon as usual.
- Test that the movie search details are cleared when closing the dialog.

## Checklist

- [x] **Quality**: This PR builds and passes the respective npm run test and works locally
- [ ] **Tests**: This PR includes thorough tests
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: node_modules/.package-lock.json
Comment (chrispinkney) [inline node_modules/.package-lock.json:1]: What is this file? I don't think we need it

Comment (chrispinkney) [inline node_modules/.package-lock.json:1]: What is this file? I don't think we need it?

File: node_modules/lodash.debounce/LICENSE
Comment (chrispinkney) [inline node_modules/lodash.debounce/LICENSE:1]: same

File: node_modules/lodash.debounce/README.md
Comment (chrispinkney) [inline node_modules/lodash.debounce/README.md:1]: why is node_modules even being committed? odd

File: node_modules/lodash.debounce/index.js
Comment (chrispinkney) [inline node_modules/lodash.debounce/index.js:1]: same

File: node_modules/lodash.debounce/package.json
Comment (chrispinkney) [inline node_modules/lodash.debounce/package.json:1]: same

File: package-lock.json
Comment (chrispinkney) [inline package-lock.json:1]: same

File: web/public/tmdb-primary-long.svg
Comment (chrispinkney) [inline web/public/tmdb-primary-long.svg:1]: For the attribution I think? Cool!

File: web/src/components/List/MovieCard/MovieCard.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			}
+		>
+			<Grid container direction="row" spacing="20px" height={87}>
+				<Grid>
+					<Box width={46} height={69} bgcolor="grey">
+						{tmdbMovie.poster_path && (
+							// This needs to be an <img/> component as we do not know the source at compile time
+							// Next complains about <img/>, so the linter needs to be disabled to avoid these warnings
+							// eslint-disable-next-line @next/next/no-img-element
+							<img
```
Comment (chrispinkney) [inline web/src/components/List/MovieCard/MovieCard.tsx:37]: Does this need to be img instead of using nextjs' Image function? just asking, I have no idea

Comment (chrispinkney) [reply web/src/components/List/MovieCard/MovieCard.tsx:37]: also eslint disables make baby jesus cry but its fine

File: web/src/components/List/MovieListTable/GenreDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+'use client';
```
Comment (chrispinkney) [inline web/src/components/List/MovieListTable/GenreDisplay.tsx:1]: does this need to be client based?

File: web/src/components/List/MovieListTable/MovieAddModal.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+	return (
+		<Modal open={open} onClose={handleClose}>
+			<Box
+				sx={{
+					position: 'fixed',
+					top: '50%',
+					left: '50%',
+					transform: 'translate(-50%, -50%)',
+					// height: 660,
```
Comment (chrispinkney) [inline web/src/components/List/MovieListTable/MovieAddModal.tsx:74]: does it need this comment?

File: web/src/components/List/MovieListTable/MovieListTable.tsx [hunk 4]
```diff
@@ -72,6 +67,7 @@ const MovieListTable = () => {
 
 	const saveList = async () => {
 		setLoading(true);
+		console.dir(movieListTableRows);
```
Comment (chrispinkney) [inline web/src/components/List/MovieListTable/MovieListTable.tsx:70]: remove please 😛

File: web/src/utils/tmdb-api.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+export const searchTmdbMovies = async (searchText: string) => {
+	console.log(searchText);
```
Comment (chrispinkney) [inline web/src/utils/tmdb-api.ts:2]: yo quiero taco bell

File: web/src/utils/tmdb-api.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+		{
+			method: 'GET',
+			headers: {
+				accept: 'application/json',
+				Authorization: `Bearer ${process.env.NEXT_PUBLIC_TMDB_TOKEN}`,
+			},
+		},
+	);
+	const movieSearchResults = await res.json();
+	console.dir(movieSearchResults);
```
Comment (chrispinkney) [inline web/src/utils/tmdb-api.ts:20]: one more

File: web/src/utils/tmdb-api.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+	return movieSearchResults;
+};
+
+export const fetchTmdbMovie = async (movieId: number) => {
+	console.log(movieId);
```
Comment (chrispinkney) [inline web/src/utils/tmdb-api.ts:25]: und here

File: web/src/utils/tmdb-api.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+	const res = await fetch(`https://api.themoviedb.org/3/movie/${movieId}`, {
+		method: 'GET',
+		headers: {
+			accept: 'application/json',
+			Authorization: `Bearer ${process.env.NEXT_PUBLIC_TMDB_TOKEN}`,
+		},
+	});
+	const movieDetails = await res.json();
+	console.dir(movieDetails);
```
Comment (chrispinkney) [inline web/src/utils/tmdb-api.ts:34]: c'est la vie

**Review by chrispinkney — CHANGES_REQUESTED** [review]
Amazing, amazing work. Truly amazing. This is my favourite thing that's been added to the app so far.

File: node_modules/.package-lock.json
Comment (Thinbishop) [reply node_modules/.package-lock.json:1]: fixed all dependencies

File: web/src/components/List/MovieListTable/GenreDisplay.tsx (hunk 2 above)
Comment (Thinbishop) [reply web/src/components/List/MovieListTable/GenreDisplay.tsx]: fixed

File: web/src/components/List/MovieListTable/MovieAddModal.tsx (hunk 3 above)
Comment (Thinbishop) [reply web/src/components/List/MovieListTable/MovieAddModal.tsx]: removed

File: web/src/components/List/MovieListTable/MovieListTable.tsx (hunk 4 above)
Comment (Thinbishop) [reply web/src/components/List/MovieListTable/MovieListTable.tsx]: removed all console logs

**Review by chrispinkney — APPROVED** [review]
Finally, perfection achieved

**Pull request merged**